### PR TITLE
upgrade to pybind11 2.11.1

### PR DESCRIPTION
**Pull request by tigercosmos**

### CODE REVIEW

**Comment by tigercosmos** [comment]
there is CI failed, but seems it's CI bug?

**Comment by j8xixo12** [comment]
> there is CI failed, but seems it's CI bug?

It seems homebrew fail to link python3.12 without any reason...

Is possible to trigger GitHub actions again for this PR ?

https://github.com/solvcon/modmesh/actions/runs/7196249792/job/19600683707?pr=264#step:4:754

**Comment by tigercosmos** [comment]
@yungyuc  finally, I make it works.
update pybind11 to 2.11.1, also update macOS CI to make github action works.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -165,12 +165,8 @@ jobs:
 
       - name: dependency by homebrew
         run: |
-          # Reopen for runner image v20231029.1
-          # old homebrew/core cause brew link and unlink not working
-          brew update  # No update because it is slow
-          # Force using python 3.11 from homebrew
-          brew unlink python
-          brew link --force --overwrite python@3.11
+          export HOMEBREW_NO_AUTO_UPDATE=1
```
Comment (yungyuc) [inline .github/workflows/lint.yml:168]: Wow love it.  This know-how needs to be kept among devs!

File: .github/workflows/nouse_install.yml [hunk 2]
```diff
@@ -100,8 +100,8 @@ jobs:
     strategy:
       matrix:
         # https://github.com/actions/runner-images/blob/main/images/macos/macos-11-Readme.md
-        # https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md
-        os: [macos-12]
+        # https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md
```
Comment (yungyuc) [inline .github/workflows/nouse_install.yml:103]: We don't need the macos-11-Readme.md link.  Let's remove it in the future.

**Review by yungyuc — APPROVED** [review]
LGTM.  Bravo.

File: .github/workflows/nouse_install.yml (hunk 2 above)
Comment (tigercosmos) [reply .github/workflows/nouse_install.yml:103]: sure

File: .github/workflows/lint.yml (hunk 1 above)
Comment (tigercosmos) [reply .github/workflows/lint.yml:168]: yeah, I found that brew will do update automatically lol

**Pull request merged**